### PR TITLE
Add Repositories using Github Apps

### PR DESCRIPTION
**Pull request by bryanjos**
connect #205
connect #96 

#### Description: 
This PR changes the way repos are added to incentivize. Instead of having a user manually add them, they would interact with the GitHub app to add and remove repos. This is per organization.

No more manually adding webhooks to repos

screen shot 2018-09-20 at 12 45 40 pm

These changes are based off of research into how Travis-CI and Coveralls handles things in their UI.
The UI above definitely could be better.
 
Installations of GitHub apps are tracked and repositories are associated with installations. When a GitHub app is uninstalled, the repos are soft deleted from incentivize

This PR requires that an OAuth app, along with a GitHub app is created for Incentivize. This is because we need an OAuth app mainly for getting the organizations a user is in.

Other than that there is some refactoring as well as removal of unneeded code.

#### Reviewer don't-forgets:

- [ ] Test coverage feels appropriate, given potential risk
- [ ] We're not doubling down on already-bad code
- [ ] If there are web UI changes, they don't add anything that could be considered client-side page navigation (unless pre-approved as being necessary by another engineer)
- [ ] If there are web UI changes, they don't add any AJAX form submits (unless pre-approved as being necessary by another engineer)
- [ ] If there are any lint rules disabled, they are disabled per-line, and were (in the reviewer's judgment) appropriate to disable
- [ ] Any new environment variables used in the app are both documented and have been added to both staging and production environments already
- [ ] Potential race conditions are either inconsequential, or the code prevents them from occurring.
- [ ] If this is a UI change that called for screenshots/GIFs, in the reviewer's judgement, they were included

### CODE REVIEW

File: test/incentivize/repos/repositories_test.exs [hunk 1]
```diff
@@ -9,7 +9,9 @@ defmodule Incentivize.Repositories.Test do
       Repositories.create_repository(%{
         "owner" => "octocat",
         "name" => "Hello-World",
-        "created_by_id" => user.id
+        "created_by_id" => user.id,
+        "public" => true,
+        "installation_id" => 12345
```
Comment (lintomat) [inline test/incentivize/repos/repositories_test.exs]: Large numbers should be written with underscores: 12_345

File: test/incentivize/repos/repositories_test.exs [hunk 2]
```diff
@@ -22,15 +24,17 @@ defmodule Incentivize.Repositories.Test do
     {:ok, repository} =
       Repositories.update_repository(repo, %{
         "owner" => "octocat",
-        "name" => "Hello-World"
+        "name" => "Hello-World",
+        "public" => true,
+        "installation_id" => 12345
```
Comment (lintomat) [inline test/incentivize/repos/repositories_test.exs]: Large numbers should be written with underscores: 12_345

File: test/incentivize_web/controllers/github_webhook_controller_test.exs [hunk 3]
```diff
@@ -2,29 +2,23 @@ defmodule IncentivizeWeb.GithubWebhookController.Test do
...
 
-    conn =
-      conn
-      |> put_req_header("content-type", "application/json")
-      |> put_req_header("x-hub-signature", "sha1=5c1c013df6e414af83d6d378d286b437592834ba")
-      |> put_req_header("x-github-event", "ping")
-      |> post(github_webhook_path(conn, :handle_webhook), json)
-
-    assert response(conn, 403)
+    "sha1=" <> (:crypto.hmac(:sha, secret, json) |> Base.encode16(case: :lower))
```
Comment (lintomat) [inline test/incentivize_web/controllers/github_webhook_controller_test.exs]: Pipe chain should start with a raw value.

File: test/support/factory.ex [hunk 4]
```diff
@@ -19,7 +19,9 @@ defmodule Incentivize.Factory do
       name: "test#{System.unique_integer([:positive])}",
       owner: "test#{System.unique_integer([:positive])}",
       webhook_secret: "12345",
-      created_by: build(:user)
+      created_by: build(:user),
+      public: true,
+      installation_id: 12345
```
Comment (lintomat) [inline test/support/factory.ex]: Large numbers should be written with underscores: 12_345

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.prod.revelry.net/relint/revelrylabs/incentivize/207).

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.prod.revelry.net/relint/revelrylabs/incentivize/207).

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.prod.revelry.net/relint/revelrylabs/incentivize/207).

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.prod.revelry.net/relint/revelrylabs/incentivize/207).

**Comment by bryanjos** [comment]
@jwietelmann for `2`, the only time that could happen is if we don't receive the webhooks for some reason. 

We do have the ability to get all of the apps installations. We could make a background task to check that the installations we have are correct and do any syncing needed.

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.prod.revelry.net/relint/revelrylabs/incentivize/207).

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.prod.revelry.net/relint/revelrylabs/incentivize/207).

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.prod.revelry.net/relint/revelrylabs/incentivize/207).

**Comment by lintomat** [comment]
To relint, [click here](http://lintron.prod.revelry.net/relint/revelrylabs/incentivize/207).

**Pull request merged**